contracts/VaultV1.sol

```diff
@@ -67,22 +57,18 @@ contract VaultV1 is ERC4626, Ownable {
     function beforeWithdraw(uint256 amount) internal virtual override {
         // require the user has atleast this much amount pending for withdraw
         // require the users unlock time is in the past
-        require(
-            requestedWithdraws[msg.sender] >= amount && unlockTime[msg.sender] <= block.timestamp,
-            "withdraw locked"
-        );
-
-        // check how much underlying we have "on hand"
-        uint256 startUnderlying = UNDERLYING.balanceOf(address(this));
+        require(unlockTime[msg.sender] <= block.timestamp, "withdraw locked");
+        require(requestedWithdraws[msg.sender] >= amount, "insufficient requested amount");
 
-        if (startUnderlying < amount && STRATEGY.withdrawable() >= amount) {
-            // not enough on hand but enough in the strategy. withdraw
-            STRATEGY.withdraw(amount);
-        } else if (startUnderlying < amount) {
-            // not enough on hand. Not enough in strategy. Revert to be safe.
+        // all funds are stored in strategy. See how much can be pulled
+        if (strategy.withdrawable() < amount) {
+            // we don't have enough funds
             revert("not enough funds in vault");
         }
 
+        // pull funds from strategy so they can be returned to the user
+        strategy.withdraw(amount);
+
         // update the users requested withdraw status
         requestedWithdraws[msg.sender] = 0;
         unlockTime[msg.sender] = 0;
```

Comment:
```suggestion
        // update the users requested withdraw status
        requestedWithdraws[msg.sender] = 0;
        unlockTime[msg.sender] = 0;

        // pull funds from strategy so they can be returned to the user
        strategy.withdraw(amount);
```
Interaction before effects. Not sure on trust assumptions for strategy, but it seems like this can be switched to bottom of function without any issues.